src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+					_namedResources.Add(name, resource);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Whether this resource should be lazily initialized. 
+		/// </summary>
+		private bool ShouldLazyInitializeResource(XamlObjectDefinition resource)
+		{
+			var typeName = resource.Type.Name;
+
+			return
+				// Styles are lazily initialized for perf considerations
+				typeName == "Style"
+				// All resources not in a top-level dictionary (ie FrameworkElement.Resources and Application.Resources declarations) are lazily
+				// initialized, this is to be able to handle lexically-forward references correctly. (In top-level dictionaries, this is already
+				// handled by creating lazy static properties for each resource.)
+				|| !_isTopLevelDictionary;
+		}
+
+		/// <summary>
+		/// Wrap ResourceDictionary resource in a lazy-initializer function.
+		/// </summary>
+		private IDisposable BuildLazyResourceInitializer(IIndentedStringBuilder writer)
+		{
+			writer.AppendLineInvariant("(global::Windows.UI.Xaml.ResourceDictionary.ResourceInitializer)(() => ");
+
+			var indent = writer.Indent();
+
+			return new DisposableAction(() =>
+			{
+				indent.Dispose();
+				writer.AppendLineInvariant(")");
+			});
+		}
+
+		/// <summary>
+		/// Populate MergedDictionaries property of a ResourceDictionary.
+		/// </summary>
+		private void BuildMergedDictionaries(IIndentedStringBuilder writer, XamlMemberDefinition mergedDictionaries, bool isInInitializer)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			BuildDictionaryCollection(writer, mergedDictionaries, isInInitializer, propertyName: "MergedDictionaries", isDict: false);
+		}
+
+		/// <summary>
+		/// Populate ThemeDictionaries property of a ResourceDictionary.
+		/// </summary>
+		private void BuildThemeDictionaries(IIndentedStringBuilder writer, XamlMemberDefinition themeDictionaries, bool isInInitializer)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			BuildDictionaryCollection(writer, themeDictionaries, isInInitializer, propertyName: "ThemeDictionaries", isDict: true);
+		}
+
+		/// <summary>
+		/// Build a collection of ResourceDictionaries.
+		/// </summary>
+		private void BuildDictionaryCollection(IIndentedStringBuilder writer, XamlMemberDefinition dictionaries, bool isInInitializer, string propertyName, bool isDict)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			if (dictionaries == null)
+			{
+				return;
+			}
+
+			if (isInInitializer)
+			{
+				writer.AppendLineInvariant("{0} = {{", propertyName);
+			}
+			else
+			{
+				writer.AppendLineInvariant("// {0}", propertyName);
+			}
+			foreach (var dictObject in dictionaries.Objects)
+			{
+				if (dictObject.Members.Count == 0)
+				{
+					continue;
+				}
+
+				var source = dictObject.Members.FirstOrDefault(m => m.Member.Name == "Source");
+				if (source != null && dictObject.Members.Any(m => m.Member.Name == "_UnknownContent"))
+				{
+					throw new Exception("Local values are not allowed in resource dictionary with Source set");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)